### PR TITLE
Feature/input testing

### PR DESCRIPTION
**Pull request by alexthompson207**
## Is this a fix or a feature?

This a feature.

## What is the change?

Add user flow tests for a user login into the app with Cypress, happy and sad paths for all user inputs

## What does it fix?

N/a.

## Where should the reviewer start?

In the `hitch_login_spec.js` look at all tests for different login views and user flows

## How should this be tested?

locally run `npm run cypress` all 46 tests should pass.

### CODE REVIEW

File: cypress/integration/hitch-login_spec.js [hunk 1]
```diff
@@ -1,6 +1,605 @@
...
+  });
+
+  it('should allow users to select a day input', () => {
+    cy.get('[type="checkbox"]').eq(0).check({ force: true }).and('have.value', 'sunday')
+  });
+
+  it('should allow users to select any number of day(s) input', () => {
+    cy.get('[type="checkbox"]').eq(1).check({ force: true }).and('have.value', 'monday')
+    cy.get('[type="checkbox"]').eq(5).check({ force: true }).and('have.value', 'friday')
+    cy.get('[type="checkbox"]').eq(6).check({ force: true }).and('have.value', 'saturday')
```
Comment (PaigeVannelli) [inline cypress/integration/hitch-login_spec.js:567]: This is interesting. I haven't seen the force: true before

File: cypress/integration/hitch-login_spec.js [hunk 2]
```diff
@@ -1,6 +1,605 @@
...
+      .should('have.value', 'billy@')
+    cy.get('input').eq(3)
+      .type('billyB')
+      .should('have.value', 'billyB')
+
+    cy.get('[type="submit"]').click()
+    cy.get('input:invalid').eq(0).should('have.length', 1)
+    cy.get('input').eq(2).then(($input) => {
+      expect($input[0].validationMessage).to.eq('Please enter a part following \'@\'. \'billy@\' is incomplete.')
+    });
```
Comment (itsnameissteven) [inline cypress/integration/hitch-login_spec.js:199]: what is the $input doing?

File: cypress/integration/hitch-login_spec.js [hunk 3]
```diff
@@ -1,6 +1,605 @@
...
+    cy.get('.register-form__header').contains('Origin and Destination');
+    cy.get('input').should('have.length', 2);
+  });
+
+  it('should not be able to click \'next\' button and move to next page if origin input is missing', () => {
+
+    cy.get('[type="submit"]').click()
+    cy.get('input:invalid').eq(0).should('have.length', 1)
+    cy.get('input').eq(0).then(($input) => {
+      expect($input[0].validationMessage).to.eq('Please fill out this field.')
```
Comment (PaigeVannelli) [inline cypress/integration/hitch-login_spec.js:481]: nice job targeting the validation message

**Review by PaigeVannelli — APPROVED** [review]

**Review by itsnameissteven — APPROVED** [review]
Wow that's really thorough. Thanks for tackling this!

**Pull request merged**